newIDE/app/src/EventsSheet/EventsTree/helpers.js

```diff
@@ -0,0 +1,92 @@
+import { type SortableTreeNode } from '.';
+
+export type MoveFunctionArguments = {
+  targetNode: SortableTreeNode,
+  node: SortableTreeNode,
+};
+
+export const moveEventToEventsList = ({
+  targetEventsList,
+  movingEvent,
+  initialEventsList,
+  toIndex,
+}: {
+  targetEventsList: gdEventsList,
+  movingEvent: ?gdBaseEvent,
+  initialEventsList: gdEventsList,
+  toIndex: number,
+}) => {
+  if (!movingEvent) return;
+  initialEventsList.moveEventToAnotherEventsList(
+    movingEvent,
+    targetEventsList,
+    toIndex
+  );
+};
+
+export const moveNodeAsSubevent = ({
+  targetNode,
+  node,
+}: MoveFunctionArguments) => {
+  if (!targetNode.event) return;
+  moveEventToEventsList({
+    targetEventsList: targetNode.event.getSubEvents(),
+    movingEvent: node.event,
+    initialEventsList: node.eventsList,
+    toIndex: 0,
+  });
+};
+
+export const moveNodeBelow = ({ targetNode, node }: MoveFunctionArguments) => {
+  if (!targetNode.event) return;
+  const toIndex =
+    node.indexInList < targetNode.indexInList && node.depth === targetNode.depth
```

Comment:
Why is there a check on the depth?

Reply:
Is this because the tree is flattened?

Reply:
IndexInList is the index of the event in the `EventsList`.
If the nodes are of same depth, we have to take into account the shift in index that comes from moving the node if the targeted node is after the dragged one. If the nodes are of different depth, there is no shift in index.
And depth check makes indexInList comparison relevant

Reply:
Alright, I think I got the logic here.

Reply:
Ubernit:

> And depth check makes indexInList comparison relevant

Prefer to start your condition with node.depth === targetNode.depth, which conveys the idea to the reader "**in the case of the same depth**, then look at this, otherwise do as usual". Otherwise the reader can be confused: "why are we testing this whereas this is not relevant... ah ok we check if it's relevant just after".